src/model/encoding/convertFromRawToDraftState.js

```diff
@@ -13,60 +13,228 @@
 
 'use strict';
 
+import type {BlockMap} from 'BlockMap';
+import type {BlockNodeConfig} from 'BlockNode';
+import type CharacterMetadata from 'CharacterMetadata';
+import type {RawDraftContentBlock} from 'RawDraftContentBlock';
 import type {RawDraftContentState} from 'RawDraftContentState';
 
-var ContentBlock = require('ContentBlock');
-var ContentState = require('ContentState');
-var DraftEntity = require('DraftEntity');
-var Immutable = require('immutable');
+const ContentBlock = require('ContentBlock');
+const ContentBlockNode = require('ContentBlockNode');
+const ContentState = require('ContentState');
+const DraftEntity = require('DraftEntity');
+const invariant = require('invariant');
+const Immutable = require('immutable');
 
-var createCharacterList = require('createCharacterList');
-var decodeEntityRanges = require('decodeEntityRanges');
-var decodeInlineStyleRanges = require('decodeInlineStyleRanges');
-var generateRandomKey = require('generateRandomKey');
+const createCharacterList = require('createCharacterList');
+const decodeEntityRanges = require('decodeEntityRanges');
+const decodeInlineStyleRanges = require('decodeInlineStyleRanges');
+const generateRandomKey = require('generateRandomKey');
 
-var {Map} = Immutable;
+const {List, Map} = Immutable;
 
-function convertFromRawToDraftState(
-  rawState: RawDraftContentState,
-): ContentState {
-  var {blocks, entityMap} = rawState;
+const decodeBlockNodeConfig = (
+  block: RawDraftContentBlock,
+  entityMap: *,
```

Comment:
Nit - eventually we may want to have a type for this. It could be that it will be easier to type after the entity API migration is finished.

Reply:
We should def add that after the migration, ATM we are using "any" for it all over the codebase, could be part of the migration task